torch/distributed/_composable/replicate.py

```diff
@@ -23,62 +25,49 @@ def replicate(
         >>> replicate(module)
     """
     torch._C._log_api_usage_once("torch.distributed.replicate")
-    _ReplicateState().mark_module(module, **kwargs)
+    _ReplicateState(ignored_modules=ignored_modules).mark_module(module, **kwargs)
     return module
 
 
-def _can_compose(module: nn.Module) -> bool:
-    r"""Check if module is composable for `replicate` API."""
-    return "fully_shard" not in _get_registry(module)
-
-
 class _ReplicateState:
-    def __init__(self) -> None:
+    def __init__(self, ignored_modules) -> None:
         self.module = None
         self.has_initialized: bool = False
         self._param_list: nn.ParameterList = nn.ParameterList()
         self.kwargs: dict = {}
+        self.ignored_modules = (
+            set(ignored_modules) if ignored_modules is not None else set()
+        )
 
     def mark_module(self, module: nn.Module, **kwargs) -> None:
-        if not _can_compose(module):
```

Comment:
any reason for removing this?

We will remove this eventually, but curious about does this prevent integrating with trec_shard?

Reply:
Actually we can keep this, because _can_compose only checked whether replicate can wrap a fully_shard which is unsupported at the moment.